test/testenv/secretutil.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2018-2021 Splunk Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testenv
+
+import (
+	b64 "encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+//SecretResponse Secret object struct
+type SecretResponse struct {
+	HecToken     string `json:"hec_token"`
+	IdxcSecret   string `json:"idxc_secret"`
+	Pass4SymmKey string `json:"pass4SymmKey"`
+	Password     string `json:"password"`
+	ShcSecret    string `json:"shc_secret"`
+}
+
+// DecodeBase64 decodes base64 and returns string
+func DecodeBase64(str string) string {
+	out, err := b64.StdEncoding.DecodeString(str)
+	if err != nil {
+		logf.Log.Error(err, "Failed to decode", "string", str)
+		return ""
+	}
+	return string(out)
+}
+
+// EncodeBase64 Encodes base64 and returns string
+func EncodeBase64(str string) string {
+	out := b64.StdEncoding.EncodeToString([]byte(str))
+	return out
+}
+
+// GetSecretObject Gets the secret object
+func GetSecretObject(deployment *Deployment, ns string) *SecretResponse {
+	secretObjectName := fmt.Sprintf(SecretObject, ns)
+	output, err := exec.Command("kubectl", "get", "secret", secretObjectName, "-n", ns, "-o", "jsonpath='{.data}'").Output()
+	if err != nil {
+		cmd := fmt.Sprintf("kubectl get pods -n %s", ns)
```

Comment:
Shouldn't it be 'kubectl get secret' instead of 'kubectl get pods'?

Reply:
Modified